### PR TITLE
[Docs]: Fix iceberg-links on docs with latest url

### PR DESCRIPTION
**Pull request by nk1506**
Few of the URLs have been outdated. I updated with the corresponding URLs from Iceberg home page.

### CODE REVIEW

File: site/docs/tools/iceberg/hive.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ Whereby the above properties are explained as below:
 
 ## Writing and reading tables
 
-To read and write into tables that are managed by Iceberg and Nessie, typical Hive SQL queries can be used. Refer to this documentation [here](https://iceberg.apache.org/hive/#querying-with-sql) for more information.
+To read and write into tables that are managed by Iceberg and Nessie, typical Hive SQL queries can be used. Refer to this documentation [here](https://iceberg.apache.org/docs/nightly/hive/#create-external-table-overlaying-an-existing-iceberg-table) for more information.
```
Comment (nk1506) [inline site/docs/tools/iceberg/hive.md:64]: Couldn't find the similar url https://iceberg.apache.org/hive/#querying-with-sql .

File: site/docs/develop/index.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ creators.
 The Iceberg format relies on a set of metadata files stored with (or near) the actual
 data tables. This allows Iceberg to fulfill the same role as the Hive Metastore for transactions without the need for
 expensive metadata scans or centralized planning (see [Iceberg
-performance](https://iceberg.apache.org/performance/)). This includes
+performance](https://iceberg.apache.org/docs/nightly/performance/)). This includes
```
Comment (adutra) [inline site/docs/develop/index.md:18]: I'm on the fence with nightly URLs, how about `https://iceberg.apache.org/docs/latest/performance/`?

File: site/docs/tools/iceberg/hive.md (hunk 1 above)
Comment (adutra) [reply site/docs/tools/iceberg/hive.md:64]: That anchor has been deleted in 2022 :-D 

https://github.com/apache/iceberg/commit/dd5b35067baac5eb004141d51d32251d2f8e808c 

The link now should be: https://iceberg.apache.org/docs/latest/hive/#dml-commands.

File: site/docs/tools/iceberg/index.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Overview
 
 Nessie works seamlessly with Iceberg in Spark3. Nessie is implemented as a custom Iceberg
-[catalog](http://iceberg.apache.org/custom-catalog/) and therefore supports all features available to any Iceberg
+[catalog](https://iceberg.apache.org/docs/latest/custom-catalog/) and therefore supports all features available to any Iceberg
 client. This includes Spark structured streaming, Presto, Trino, Flink and Hive. See the [Iceberg docs](https://iceberg.apache.org)
```
Comment (adutra) [inline site/docs/tools/iceberg/index.md:5]:
```suggestion
client. This includes Spark structured streaming, Presto, Trino, Flink and Hive. See the [Iceberg docs](https://iceberg.apache.org/docs/latest/)
```

**Review by adutra — APPROVED** [review]

**Pull request merged**